Userland/Libraries/LibJS/Runtime/AbstractOperations.cpp

```diff
@@ -574,4 +575,89 @@ Value canonical_numeric_index_string(GlobalObject& global_object, PropertyName c
     return n;
 }
 
+// 22.1.3.17.1 GetSubstitution ( matched, str, position, captures, namedCaptures, replacement ), https://tc39.es/ecma262/#sec-getsubstitution
+String get_substitution(VM& vm, GlobalObject& global_object, String const& matched, String const& str, size_t position, Vector<Value> const& captures, Value const& named_captures, Value const& replacement)
```

Comment:
We generally do not pass `vm` to abstract operations, and get it via the global object:
```c++
auto& vm = global_object.vm();
```